api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -232,7 +233,27 @@ message TlsSessionTicketKeys {
       [(validate.rules).repeated = {min_items: 1}, (udpa.annotations.sensitive) = true];
 }
 
-// [#next-free-field: 13]
+// Indicates a certificate to be obtained from a named CertificateProvider plugin instance.
+// The plugin instances are defined in the client's bootstrap file.
+// The plugin allows certificates to be fetched/refreshed over the network asynchronously with
+// respect to the TLS handshake.
+// [#not-implemented-hide:]
+message CertificateProviderPluginInstance {
```

Comment:
Apart from moving from tls.proto it seems we have also renamed it from `CertificateProviderInstance`. Are we forcing all certificate providers to be plugins thru this nomenclature? Also I think "extension" is preferred over "plugin" IIRC from a previous PR/comment but I don't share that preference.